internal/venice-common/src/main/java/com/linkedin/venice/pushmonitor/ReplicaStatus.java

```diff
@@ -47,14 +47,16 @@ public ReplicaStatus(String instanceId, boolean enableStatusHistory) {
     this.statusHistory = enableStatusHistory ? new LinkedList<>() : null;
   }
 
-  public void updateStatus(ExecutionStatus newStatus) {
-    currentStatus = newStatus;
+  public void updateStatus(ExecutionStatus newStatus, String incrementalPushVersion) {
+    setIncrementalPushVersion(incrementalPushVersion);
+    if (!isIncrementalPushStatus(newStatus)) {
+      setCurrentStatus(newStatus);
+    }
     addHistoricStatus(newStatus);
```

Comment:
I assume we should pass `incrementalPushVersion` to this function, and right now, it is using the `incrementalPushVersion` updated by this function, but it seems very error prone, and passing it via param is more robust.